object_detection/utils/box_label_manager.py

```diff
@@ -0,0 +1,195 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""This class manages the location of labels over an image.
+
+The objective is to minimize overlapping. The current algo is pretty simple but covers
+the majority of cases.
+"""
+
+# Max overlapping = 20%
+MAX_COVERING = 0.2
+
+class BoxLabelManager:
+
+    label_bounding_boxes = []
+    bounding_boxes       = []
+    image_width  = 0
+    image_height = 0
+    draw_rectangle = False
+    draw_label     = False
+
+    def __init__(self, _image_width, _image_height):
+        self.image_width  = _image_width
+        self.image_height = _image_height
+        self.label_bounding_boxes = []
+        self.bounding_boxes       = []
+
+    def setDrawRectangle(self, _value):
+        self.draw_rectangle = _value;
+
+    def isDrawRectangle(self):
+        return self.draw_rectangle
+
+    def isDrawLabel(self):
+        return self.draw_label
+
+    def setDrawLabel(self, _value):
+        self.draw_label = _value;
+
+    #
+    def addBoundingBoxAndLabelBox(self, _bounding_box, _label_width, _label_height):
+        """" Return the coordinates of the best label box and store it.
+
+                Args:
+                    _bounding_box: first box
+                    _label_width: second box
+                    _label_height:
+
+                Returns:
+                    A bounding box for the label as [left, top, right, bottom]
```

Comment:
You mean the return values are coordinates of the label box, right? If so, I suggest renaming the variables to left_x, top_y, right_x, bottom_y so that they are clearly to represent the coordinates. Anyway, this is not a big problem.

Reply:
Corrected